### DISCUSSION TITLE
FCM Direct channel

### DISCUSSION THREAD

**Discussion by eakoli** [firebase/firebase-ios-sdk#9260] [Pitches]
* Firebase Component: Messaging

Is there some documentation/discussion somewhere as to the rational for removal of the FCM direct channel support from iOS?

This was a very valuable feature as it allowed full testing using the iOS simulator, and provided slightly more reliable data only messaging when the application was in the foreground.

**Comment by paulb777** [comment]
Thanks for the input @eakoli .  We determined the ongoing cost of maintaining the FCM direct channel wasn't worth the value it provided based on the prevalence of APNS for IOS. Sorry that does not match with your needs.

**Reply by eakoli** [reply]
@paulb777 Can you provide the PR/commit/release where that functionality was removed ?

**Reply by charlotteliang** [reply]
Should be the major 7.0.0 release: https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseMessaging/CHANGELOG.md#2020-10----v700
